qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -100,37 +114,115 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
+									codeActions, UNDEFINED_NAMESPACE_SEVERITY_SETTING);
+							break;
+						case UnknownProperty:
+							doCodeActionToCreateProperty(template, diagnostic, errorCode, codeActions,
+									codeActionResolveFutures,
+									resolver,
+									sharedSettings);
+							break;
+						default:
+							break;
 					}
 				}
 			}
 		}
-		return CompletableFuture.completedFuture(codeActions);
+		CompletableFuture<Void>[] registrationsArray = new CompletableFuture[codeActionResolveFutures.size()];
+		codeActionResolveFutures.toArray(registrationsArray);
+		return CompletableFuture.allOf(registrationsArray).thenApply((Void _void) -> {
+			return codeActions;
+		});
+	}
+
+	private void doCodeActionToCreateProperty(Template template, Diagnostic diagnostic, QuteErrorCode errorCode,
+			List<CodeAction> codeActions, List<CompletableFuture<Void>> registrations,
+			QuteTemplateGenerateMissingJavaMember resolver, SharedSettings settings) {
+		JsonObject data = (JsonObject) diagnostic.getData();
+		if (data == null) {
+			// TODO: the client may not support diagnostic data; resolve the needed
+			// information some other way
+			return;
+		}
+		MissingMemberData missingMemberData = DiagnosticDataFactory.getMissingMemberData(data);
+		String missingProperty = missingMemberData.getProperty();
+		String resolvedType = missingMemberData.getSignature();
+		String propertyCapitalized = missingProperty.substring(0, 1).toUpperCase() + missingProperty.substring(1);
+
+		String projectUri = template.getProjectUri();
+
+		GenerateMissingJavaMemberParams publicFieldParams = new GenerateMissingJavaMemberParams(MemberType.Field,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams getterParams = new GenerateMissingJavaMemberParams(MemberType.Getter,
+				missingProperty, resolvedType, projectUri);
+		GenerateMissingJavaMemberParams templateExtensionParams = new GenerateMissingJavaMemberParams(
+				MemberType.TemplateExtension, missingProperty, resolvedType, projectUri);
+
+		CodeAction createPublicField = createCodeActionWithData(
+				String.format(CREATE_PUBLIC_FIELD, missingProperty, resolvedType), publicFieldParams,
+				Collections.singletonList(diagnostic));
+		CodeAction createGetter = createCodeActionWithData(
+				String.format(CREATE_GETTER, propertyCapitalized, resolvedType), getterParams,
+				Collections.singletonList(diagnostic));
+		CodeAction createTemplateExtension = createCodeActionWithData(String
+				.format(CREATE_TEMPLATE_EXTENSION, missingProperty),
+				templateExtensionParams, Collections.singletonList(diagnostic));
+
+		registrations.add(resolver.resolveTemplateCodeAction(publicFieldParams) //
+				.thenAccept((workspaceEdit) -> {
+					if (workspaceEdit == null) {
+						return;
+					}
+					createPublicField.setEdit(workspaceEdit);
+					synchronized (codeActions) {
+						codeActions.add(createPublicField);
+					}
+				}));
+
+		registrations.add(resolver.resolveTemplateCodeAction(getterParams) //

```

Comment:
I think it should be better that you create the codeaction without the edit and you add it in codeActions.add(createGetter

The thenApply will update the edit code action and that is it.

If you do that you will control the order of codeaction add in